### PR TITLE
chore: migrate from structopt to clap

### LINKED ISSUE #109: Remove structopt dependency and use latest clap instead
`structopt` depends on old clap version [`2.33`](https://github.com/TeXitoi/structopt/blob/master/Cargo.toml#L30). It causes a dependence of `ansi_term`. `ansi_term` is no longer [maintained](https://github.com/ogham/rust-ansi-term/issues/72). To remove a dependency to `ansi_term`, we must remove the `structopt` dependency and use latest `clap` which provides `structopt` like interface.

### PR DESCRIPTION
**Pull request by ryota-sakamoto**
*Issue #, if available:*

close #109

*Description of changes:*

structopt is now in maintenance mode so we need to migrate into clap.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by StoneDot** [comment]
This task is blocked by #143.

**Review by StoneDot — COMMENTED** [review]
Thank you for tackling this hard task. I left a few comments. Could you please look them?

File: src/cmd.rs [hunk 1]
```diff
@@ -28,129 +27,125 @@ const ABOUT_DYNEIN: &str = "\
...
 
-    #[structopt(long, required_if("child", "None"), conflicts_with("child"))]
+    #[clap(long, verbatim_doc_comment)]
     pub shell: bool,
 
     /// This option displays detailed information about third-party libraries, frameworks, and other components incorporated into dynein,    
     /// as well as the full license texts under which they are distributed.
-    #[structopt(long, required_if("child", "None"), conflicts_with("child"))]
+    #[clap(long)]
     pub third_party_attribution: bool,
```
[the remark this answers is not in the document]
Comment (ryota-sakamoto) [reply src/cmd.rs:58]: I will address the this point in another pull request.

**Review by StoneDot — APPROVED** [review]
Thank you for checking my comment. LGTM

**Pull request merged**